File: 34_FindFirstLastPos.cpp
#include<iostream>
#include<vector>
using namespace std;

class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) 
    {
        int i = 0,j = nums.size()-1;
        int n = -1,m = -1;
        while(i <= j)
        {
            if(m == -1)
            {
                if(nums[i] == target)  m = i;
                if(m == -1) i++;
            }
            if(n == -1)   
            {
                if(nums[j] == target)  n = j;
                if(n == -1) j--;
            }
            if(m != -1 && n != -1) break;
        }
        return {m,n};
    }
};

int main()
{
    // vector<data_type> array_name[array_size];  
    vector<int> vect{ 10, 20, 30 };
    int *ret = NULL;

    Solution obj;
    obj.searchRange(vect,10);

    return 0;
}

File: 34_FindFirstLastPos.js

var searchRange = function(nums, target) {
  
    let result = [];
    
    if(nums.includes(target)) {
        result.push(nums.indexOf(target));
        for(var i = nums.indexOf(target); i < nums.length; i++) {
            if(nums[i+1] != target) {
                result.push(i);
                return result;
            }
        }
    } else {
        return [-1,-1];
    }
};

let main = function() {

    let  nums = [5,7,7,8,8,10], target = 5;

    result = searchRange(nums,target);
    console.log(result);
}

main();